AndroidAnnotations/androidannotations/src/main/java/org/androidannotations/holder/EFragmentHolder.java

```diff
@@ -547,6 +549,15 @@ public JBlock getOnDetachBeforeSuperBlock() {
 	}
 
 	@Override
+	public JBlock getIntentFilterTarget(IntentFilterData intentFilterData) {
+		if (RegisterAt.OnAttachOnDetach.equals(intentFilterData.getRegisterAt())) {
+			return getOnAttachAfterSuperBlock();
+		} else {
```

Comment:
Remove the else block, return is enough. Also it maybe more flexible to return super.getIntentFilterTarget here.